catalog/schematron/catalog-schematron-plugin/src/test/java/ddf/services/schematron/SchematronValidationServiceTest.java

```diff
@@ -166,6 +167,26 @@ public void testWithSpacesInPathToSchematronFile()
                 "dog_3leg_3paw.xml"));
     }
 
+    @Test
+    public void testWarningsAndErrorsAreSanitized()
+            throws ValidationException, IOException, SchematronInitializationException {
+        SchematronValidationService service = getService(false,
+                "doggy-namespace",
+                true,
+                "dog_legs.sch",
+                "dog_paws.sch");
+        MetacardImpl metacard = getMetacard("dog_4leg_4paw_namespace.xml");
+        service.validate(metacard);
+        assertThat(service.getSchematronReport(), is(notNullValue()));
+        Pattern pattern = Pattern.compile("[\t \r\n]+");
+        for (String warning : service.getSchematronReport().getWarnings()) {
+            assertThat(pattern.matcher(warning).matches(), is(false));
```

Comment:
Updated, I changed that test to simply see if the sanitized string is the same as the exception being thrown. I also added direct tests to the sanitize method.